homeassistant/components/weheat/config_flow.py

```diff
@@ -34,11 +37,51 @@ def extra_authorize_data(self) -> dict[str, str]:
     async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
         """Override the create entry method to change to the step to find the heat pumps."""
         # get the user id and use that as unique id for this entry
-        user_id = await async_get_user_id_from_token(
-            API_URL,
-            data[CONF_TOKEN][CONF_ACCESS_TOKEN],
-            async_get_clientsession(self.hass),
-        )
+        try:
+            user_id = await async_get_user_id_from_token(
+                API_URL,
+                data[CONF_TOKEN][CONF_ACCESS_TOKEN],
+                async_get_clientsession(self.hass),
+            )
+        except ClientConnectorError as e:
+            LOGGER.error("Failed to get user id: %s", e)
+            return self.async_abort(reason="get_user_failed")
+        except Exception as e:  # noqa: BLE001
+            LOGGER.exception("Unexpected error: %s", e)
+            return self.async_abort(reason="unknown")
+
+        # Test getting heat pumps
+        try:
+            discovered_heat_pumps = await HeatPumpDiscovery.async_discover_active(
+                API_URL,
+                data[CONF_TOKEN][CONF_ACCESS_TOKEN],
+                async_get_clientsession(self.hass),
+            )
+        except ClientConnectorError as e:
+            LOGGER.error("Failed to get heat pumps: %s", e)
+            return self.async_abort(reason="get_heat_pumps_failed")
```

Comment:
I think these 2 can be combined, in case of get_user_failed or get_heat_pumps_failed, I would suggest just showing `cannot_connect`, as that is relevant for the user